ray-operator/controllers/raycluster_controller.go

```diff
@@ -262,7 +291,7 @@ func (r *RayClusterReconciler) reconcilePods(instance *rayiov1alpha1.RayCluster)
 		} else if diff == 0 {
 			log.Info("reconcilePods", "all workers already exist for group", worker.GroupName)
 			continue
-		} else if int32(len(runningPods.Items)) == (*worker.Replicas + int32(len(worker.ScaleStrategy.WorkersToDelete))) {
+		} else if -diff == int32(len(worker.ScaleStrategy.WorkersToDelete)) {
```

Comment:
`worker.ScaleStrategy.WorkersToDelete` has been set to 0 in line 274 if the flag is true. I think you may want to assign the value to a different variable?